be/src/exprs/vectorized/math_functions.cpp

```diff
@@ -439,22 +434,46 @@ ColumnPtr MathFunctions::conv_string(FunctionContext* context, const starrocks::
             result.append_null();
             continue;
         }
-
-        StringParser::ParseResult parse_res;
-        int64_t decimal_num = StringParser::string_to_int<int64_t>(reinterpret_cast<char*>(string_value.data),
-                                                                   string_value.size, src_base_value, &parse_res);
-
-        if (src_base_value < 0 && decimal_num >= 0) {
-            result.append_null();
+        bool is_signed = src_base_value < 0;
+        char* data_ptr = reinterpret_cast<char*>(string_value.data);
+        int digit_start_offset = StringParser::skip_leading_whitespace(data_ptr, string_value.size);
+        if (digit_start_offset == string_value.size) {
+            result.append(Slice("0", 1));
             continue;
         }
-
-        if (!handle_parse_result(dest_base_value, &decimal_num, parse_res)) {
+        bool negative = data_ptr[digit_start_offset] == '-';
+        digit_start_offset += negative;
+        StringParser::ParseResult parse_res;
+        uint64_t decimal64_num = StringParser::string_to_int<uint64_t>(data_ptr + digit_start_offset,
+                                                                       string_value.size - digit_start_offset,
+                                                                       std::abs(src_base_value), &parse_res);
+        if (parse_res == StringParser::PARSE_SUCCESS) {
+            if (is_signed) {
+                if (negative && decimal64_num > 0ull - std::numeric_limits<int64_t>::min()) {
+                    decimal64_num = 0ull - std::numeric_limits<int64_t>::min();
+                }
+                if (!negative && decimal64_num > std::numeric_limits<int64_t>::max()) {
+                    decimal64_num = std::numeric_limits<int64_t>::max();
+                }
+            }
+        } else if (parse_res == StringParser::PARSE_FAILURE) {
```

Comment:
Could this branch merge with the branch `else if (parse_res == StringParser::PARSE_OVERFLOW)` if type of decimal64_num is `int64_t`?

Reply:
Yes, these two branches are both dealing with the overflow problem, but merging them will not make the implementation more concise because we can't merge the overflow check condition, so I'd prefer to keep the status quo. what do you think? @liuyehcf